engines/e_padlock.c

```diff
@@ -7,6 +7,12 @@
  * https://www.openssl.org/source/license.html
  */
 
+/*
+ * This file uses the low level AES functions (which are deprecated for
+ * non-internal use) in order to implement the padlock engine AES ciphers.
+ */
+#define OPENSSL_SUPPRESS_DEPRECATED
```

Comment:
@mattcaswell does this work for an engine?
Won't the low level calls be hidden and non-resolvable?

Reply:
There are two scenarios:

1) Compilation when "no-deprecated" has been defined

2) Compilation when "no-deprecated" has not been defined

In the first case this PR changed Configure so that padlock never gets compiled. In the second case the low level calls are resolvable and we are only concerned about the deprecation warnings that the compiler will emit. This line is to suppress those.

Reply:
Thanks for the explanation, I missed the Configure change somehow.

Is this how we want to deal with engines that use deprecated APIs?  e.g. the e_dasync engine uses SHA-1 and I modified it to not over in #10791.

Reply:
In general no. I think where possible we should convert usage to use EVP instead. I took the decision not to do that with padlock because the usage was so extensive.